controllers/admin/AdminImportController.php

```diff
@@ -3010,6 +3010,21 @@ protected function attributeImportOne($info, $default_language, &$groups, &$attr
                     StockAvailable::setQuantity((int) $product->id, $id_product_attribute, (int) $info['quantity'], $this->context->shop->id);
                 }
             }
+
+            // assign combination id to already associated product suppliers
+            $productSuppliers = ProductSupplier::getSupplierCollection($product->id);
```

Comment:
Be careful I believe this returns ALL the existing ProductSupplier including the ones associated to already existing combinations. So it seems this code works fine for a product without combinations since the only ProductSupplier returned will the one associated to the Product only.

Then cloning this and associating it to the new combinations is fine. But when you perform this import again then the ProductSupplier entities associated to other combinations will also be cloned and associated to the new combination resulting in this new combination having multiple ProductSupplier instances for the same supplier.

I maybe be missing something but I think you should only loop through the ProductSuppliers associated to product only (with `$productSupplier->id_product_attribute === 0`) meaning you will only loop through the list of suppliers associated to the product.

And also the ProductSupplier should only be cloned and associated if the combination doesn't already have an associated ProductSupplier for this supplier. I don't know if I'm clear 😅 but we can discuss this live.

Reply:
hmm.
Now Im looping through all of them and so I can check if it is already associated or not.
If it is associated, i skip it, this way only creating new ProductSupplier for the ones that are not associated with combination already.

If i would check only with `id_product_attribute === 0` then I won't be able to check if this combination already have the supplier assigned :thinking:

ofc considering performance, I guess if there are alot of combinations, then this will be costly, maybe its worth introducing a separate custom query or so?